test/javascript/components/mentoring/inbox/ConversationList.test.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import { render, screen, fireEvent, waitFor } from '@testing-library/react'
+import { rest } from 'msw'
+import { setupServer } from 'msw/node'
+import '@testing-library/jest-dom/extend-expect'
+import { ConversationList } from '../../../../../app/javascript/components/mentoring/inbox/ConversationList.jsx'
+
+const server = setupServer(
+  rest.get('https://exercism.test/conversations', (req, res, ctx) => {
+    return res(ctx.status(500, 'Internal server error'))
+  })
+)
+
+beforeAll(() => server.listen())
+afterEach(() => server.resetHandlers())
+afterAll(() => server.close())
+
+test('allow retry after loading error', async () => {
+  const setPage = jest.fn()
+
+  render(
+    <ConversationList
+      request={{
+        endpoint: 'https://exercism.test/conversations',
+        query: { page: 2 },
+      }}
+      setPage={setPage}
+    />
+  )
+
+  await waitFor(() => expect(screen.getByText('Retry')).toBeInTheDocument())
```

Comment:
@kntsoriano For some reason this does not work. I've dug into this a bit and even though I've added the following server setup:

```js
rest.get('https://exercism.test/conversations', (req, res, ctx) => {
    return res(ctx.status(500, 'Internal server error'))
  })
```

The `status` key returned by the `usePaginatedRequestQuery` call in the `ConversationList` component does not change from `loading` to `error`. Passing valid JSON data does change that status from `loading` to `success`. Any ideas?

Reply:
```
    <ConversationList
      request={{
        endpoint: 'https://exercism.test/conversations',
        query: { page: 2 },
        options: {  retry: false }
      }}
```

Can you try that?

Reply:
Ha, found it! There was a mistake in how options were passed to the `usePaginatedQuery` and `useQuery` functions. I've fixed it in this PR.